docs(authn): enrich introduction for authn (#1224)
emqtt/emq-docs-cn: en_US/security/authn/jwt.md

## Authentication Principle

> The client carries the JWT in the connection request, and EMQX uses the pre-configured secret or public key to verify the JWT signature. If the user is using JWKS, the JWT authenticator attempts to verify the JWT signature using all currently valid public keys. If the signature verification is successful, the JWT authenticator proceeds to check the claims. If there are claims such as `iat`, `nbf` or `exp`, the JWT authenticator will actively check the validity of the JWT based on these claims. In addition to this, we also allow users to specify some additional claims checks. The client is finally allowed to login only if the signature verification and claims check pass together.

Reviewer: ```suggestion
The client carries the JWT in the connection request, and EMQX uses the pre-configured secret or public key to verify the JWT signature. If the user is using an authentication servert that provides a JWKS endpoint, the JWT authenticator attempts to verify the JWT signature using all currently valid public keys. If the signature verification is successful, the JWT authenticator proceeds to check the claims. If there are claims such as `iat`, `nbf` or `exp`, the JWT authenticator will actively check the validity of the JWT based on these claims. In addition to this, we also allow users to specify some additional claims checks. The client is finally allowed to login only if the signature verification and claims check pass together.
```

## Common Usage

> The name of the JWT authenticator may be misleading, but in essence it just checks the signature of the JWT, which means that the JWT authenticator does not guarantee the legitimacy of the client's identity.

Reviewer: ```suggestion
Used in the most basic way, the JWT authenticator just checks the signature of the JWT, which means that the JWT authenticator does not guarantee the legitimacy of the client's identity.
```

Reviewer: I mean, EMQX actually _can_ verify identity, if STS provides appropriate tokens.

Author: But when the client gets the JWT, it usually means that it has been authenticated once. At most, EMQX checks whether the JWT matches the current holder.

> Therefore, the common usage is that the user deploys an independent authentication server, the client first accesses the authentication server, the authentication server verifies the identity of the client, and issues JWT for the legitimate client, and then the client uses the obtained JWT to connect to EMQX .

Reviewer: ```suggestion
The common usage is that the user deploys an independent authentication server. The client first accesses the authentication server, the authentication server verifies the identity of the client, and issues JWT for the legitimate client, and then the client uses the obtained JWT to connect to EMQX.

To provide identity verification, the authentication server should include the client's username or clientid as JWT claims, and EMQX JWT authenticator should be configured to check these claims.

```

In order to reduce the possibility of JWT leakage and theft, in addition to setting a reasonable validity period, we also recommend using TLS to encrypt client connections.

## Authorization

This is an optional function, we define a private Claim `acl`, which is used to carry the access rules of publish and subscribe in the JWT to control the permissions of the client after login.

Claim `acl` defines 3 optional fields, `pub`, `sub` and `all`, which are used to specify the whitelist of publish, subscribe and publish-subscribe topics respectively. Topic wildcards and placeholders are allowed in topic entries (currently only `${clientid}` and `${username}` are supported). Since there may be cases where topic content conflicts with placeholder syntax, we also provide the `eq` syntax to cancel placeholder interpolation. Example:

```json
{
  "exp": 1654254601,
  "username": "myuser",
  "acl": {
    "pub": [
      "testpub1/${username}",
      "eq testpub2/${username}"
    ],
    "sub": [
      "testsub1/${username}",

>       "testpub2/${clientid}"

Reviewer: Is it intended to have `testpub2` in `sub` section?

Author: sorry, this is a typo.
